mailbox/store/src/main/java/org/apache/james/mailbox/store/transaction/Mapper.java

```diff
@@ -53,15 +53,15 @@
          */
         T run() throws MailboxException;
     }
-    
-    
-    abstract class VoidTransaction implements Transaction<Void> {
-        
-        public final Void run() throws MailboxException {
-            runVoid();
-            return null;
-        }
-        public abstract void runVoid() throws MailboxException;
 
+    interface Operation {
+        void run() throws MailboxException;
+    }
+
+    static  Transaction<Void> toTransaction(Operation operation) throws MailboxException {
```

Comment:
I don't think static is needed here

Reply:
It is otherwise intelliJ reports I can not write a body block. And I don't want a default methods.

Reply:
Can you give us some rational about avoiding default methods ?

Reply:
You need an instance to use it, in this special use case.

Otherwise, I remember a discussion with you 2.5 years ago, telling the least wanted feature of Java-8. I listenend. So let's say *I have nothing personnaly against it*. It's not a team practice (yet). I don't need it here. So let's not do it.

Reply:
static method interface is not a team practice neither

Reply:
But static method in classes is.

Reply:
It' just that I don't have a relevant class at first glance, but it can very well be extracted.